analyzer/language/python/pip/pip.go

```diff
@@ -30,9 +32,24 @@ func (a pipLibraryAnalyzer) Analyze(_ context.Context, input analyzer.AnalysisIn
 	return res, nil
 }
 
-func (a pipLibraryAnalyzer) Required(filePath string, _ os.FileInfo) bool {
+func (a pipLibraryAnalyzer) Required(dir string, filePath string, osInfo os.FileInfo) bool {
 	fileName := filepath.Base(filePath)
-	return fileName == requiredFile
+	if fileName == requiredFile {
+		return true
+	}
+
+	// Test for .txt/.in file combinations, which indicate pip-compile files.
+	if dir != "" && strings.HasSuffix(fileName, ".txt") {
+		// Create new filepath that has the same path but ends with .in
+		inFilePath := strings.TrimSuffix(filePath, ".txt") + ".in"
+
+		// Check whether in file exists.
+		if _, err := os.Stat(path.Join(dir, inFilePath)); err == nil {
```

Comment:
Unfortunately, it doesn't work for container images.

Reply:
@knqyf263 Is there a way to get this to work in docker though?